### PR TITLE
COMMERCE-637 Fixed discount percentage

### PR DESCRIPTION
**Pull request by alessiorendina**

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test

**Comment by marco-leo** [comment]
@Commando18 After te change the discount tests are failing Please review your changes

**Comment by alessiorendina** [comment]
ci:retest

**Comment by marco-leo** [comment]
@Commando18 The tests relative to discounts are failing. Please review

File: commerce-discount-service/src/main/java/com/liferay/commerce/discount/internal/CommerceDiscountCalculationImpl.java [hunk 1]
```diff
@@ -278,17 +279,20 @@ private CommerceDiscountValue _getCommerceDiscountValue(
 			discountedAmount = discountedAmount.subtract(currentDiscountAmount);
 		}
 
-		BigDecimal discountPercentage = discountedAmount.divide(
-			amount, RoundingMode.valueOf(commerceCurrency.getRoundingMode()));
+		BigDecimal discountPercentage = discountedAmount.divide(amount);
```
Comment (marco-leo) [inline commerce-discount-service/src/main/java/com/liferay/commerce/discount/internal/CommerceDiscountCalculationImpl.java]: @Commando18 You have to use the rounding mode always

File: commerce-discount-service/src/main/java/com/liferay/commerce/discount/internal/CommerceDiscountCalculationImpl.java [hunk 2]
```diff
@@ -278,17 +279,20 @@ private CommerceDiscountValue _getCommerceDiscountValue(
...
 			commerceCurrency, currentDiscountAmount);
 
+		MathContext mathContext = new MathContext(
+			discountPercentage.precision(),
+			RoundingMode.valueOf(commerceCurrency.getRoundingMode()));
+
 		return new CommerceDiscountValue(
 			commerceDiscount.getCommerceDiscountId(), discountAmount,
-			_ONE_HUNDRED.subtract(discountPercentage), values);
+			_ONE_HUNDRED.subtract(discountPercentage, mathContext), values);
```
Comment (marco-leo) [inline commerce-discount-service/src/main/java/com/liferay/commerce/discount/internal/CommerceDiscountCalculationImpl.java]: No nned to use a MathContext

**Review by marco-leo — APPROVED** [review]

**Comment by alessiorendina** [comment]
ci:retest

**Pull request merged**